File: APIClothesEcommerceShop/APIClothesEcommerceShop/DTO/Statistics/ComboStatisticsResponse.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace APIClothesEcommerceShop.DTO.Statistics
{
    public class ComboStatisticsResponse
    {
        public int TotalCombos { get; set; } // Tổng số combo
        public int TotalActiveCombos { get; set; } // Tổng số combo đang hoạt động
        public int TotalInactiveCombos { get; set; } // Tổng số combo không hoạt động
        public decimal AverageComboPrice { get; set; } // Giá trung bình của combo
        public decimal TotalComboRevenue { get; set; } // Tổng doanh thu từ combo
    }
}

File: ECOMMERCESHOPUXUI/EcommerceProject/src/models/dtos/statisticsDtos/comboStatisticsResponse.js
export default class ComboStatisticsResponse {
  constructor() {
    this.totalCombos = 0 // Tổng số combo
    this.totalActiveCombos = 0 // Tổng số combo đang hoạt động
    this.totalInactiveCombos = 0 // Tổng số combo không hoạt động
    this.averageComboPrice = 0.0 // Giá trung bình của combo
    this.totalComboRevenue = 0.0 // Tổng doanh thu từ combo
  }

  static fromApiResponse(data) {
    const response = new ComboStatisticsResponse()
    response.totalCombos = data.totalCombos || 0
    response.totalActiveCombos = data.totalActiveCombos || 0
    response.totalInactiveCombos = data.totalInactiveCombos || 0
    response.averageComboPrice = data.averageComboPrice || 0.0
    response.totalComboRevenue = data.totalComboRevenue || 0.0

    return response
  }
}
